extensions/amp-story/1.0/amp-story-request-service.js

```diff
@@ -56,12 +56,33 @@ export class AmpStoryRequestService {
   /**
    * Retrieves the publisher share providers.
    * Has to be called through `loadShareConfig`.
+   * @param  {!Element} shareConfigEl
    * @return {(!Promise<!JsonObject>|!Promise<null>)}
    */
-  loadShareConfigImpl_() {
-    const shareConfigEl = this.storyElement_.querySelector(
-      'amp-story-social-share, amp-story-bookend'
-    );
+  getInlineConfig(shareConfigEl) {
+    // Fallback. Check for an inline json config.
+    let config = null;
+    try {
+      config = getChildJsonConfig(shareConfigEl);
+    } catch (err) {}
```

Comment:
Let's handle the promise rejection here and refactor this.
We can do this by adding return Promise.reject(err).

 We don't need the config variable. We can return the promise in the `try`.
```
  getInlineConfig(shareConfigEl) {
    try {
      return Promise.resolve(getChildJsonConfig(shareConfigEl));
    } catch (err) {
      return Promise.reject(err);
    }
  }
```